src/quicklist.h

```diff
@@ -46,8 +46,8 @@
 typedef struct quicklistNode {
     struct quicklistNode *prev;
     struct quicklistNode *next;
-    unsigned char *zl;
-    unsigned int sz;             /* ziplist size in bytes */
+    union {unsigned char *zl; unsigned char *entry;};
+    size_t sz;             /* entry size in bytes */
     unsigned int count : 16;     /* count of items in ziplist */
     unsigned int encoding : 2;   /* RAW==1 or LZF==2 */
     unsigned int container : 2;  /* NONE==1 or ZIPLIST==2 */
```

Comment:
In the recent past i suggested to change `QUICKLIST_NODE_CONTAINER_NONE` to `QUICKLIST_NODE_CONTAINER_PLAIN`, so that whenever we want to refer to a non-ziplist (or soon to be non-listpack) node we can just say something like this in comments: "for **plain** nodes, we do...."
i.e. we can't say "for **none** nodes...", maybe we can say "for **non-ziplist** nodes", but that's awkward.

now i wanna suggest something else: maybe we'll call them packed and unpacked?
i.e. we won't use the term ziplist or listpack, but use a general term of "packed"
```suggestion
    unsigned int encoding : 2;   /* RAW==1 or LZF==2 */
    unsigned int container : 2;  /* UNPACKED==1 or PACKED==2 */
```

in theory we can also change the above constant from 1 and 2, to 0 and 1, and then use these fields as booleans.
i.e. if we rename `int container` to `int packed`
```c
if (!node->packed)
```
but i'm not sure we wanna go that way.
@sundb what do you think?

Reply:
I think it works, and I can't think of a situation where we need to support multiple containers.

Reply:
LOL.. i wanna remind you #8880 
https://github.com/sundb/redis/blob/adfcc66787550e131878dc67442dbe3b969ae036/src/quicklist.h#L53

on a second though I think i'd rather avoid drastic changes that we may some day wanna revert.
so i stand behind my suggestion above keeping the field names, and just renaming the enums to PACKED and UNPACKED.
this way when listpacks arrive to town, we won't need to rename the ZIPLIST (PACKED can refer to either ziplist or listpack).
and also considering that the `zl` field now has a union alias named `entry` that can also play along, i.e. instead of renaming the `zl` field to `lp`, we can just delete it and use `entry`.

p.s. i'm not sure which one of these PRs will be merged first.
possibly the listpack one will be easier since it just replaces one packing format with another one, and they already have similar capabilities.
whereas this PR still needs to figure out the lzf compression complications (which don't pose any concern for the other PR).

so whichever PR is merged first, they should both use the same variable and enum names (i.e. `entry` and `CONTAINER_PACKED`)

agreed?

Reply:
Agree. It doesn't matter which PR will be merged first,

Reply:
so bottom line is that I only change in naming is : 
unsigned int container : 2;  /* NONE==1 or ZIPLIST==2 */ ==== > 
    unsigned int container : 2;  /* UNPACKED==1 or PACKED==2 */ ?

Reply:
As I understand it, it should be `unsigned int packed : 2; / UNPACKED==1 or PACKED==2 */ ? `.
Because using `node->container` is weird, it doesn't mean it's packed .

Reply:
@oranagra do you agree with that ?

Reply:
i was thinking we're gonna use the term `PLAIN` (not `UNPACKED`)?
i don't have any objection to the term `container` (listpack, and ziplist are containers, and plain means there's no container),

since `NONE` was completely unused till now, there's no reason not to rename it.
however, since we're going to rename `ZIPLIST` soon to `LISTPACK`, then i don't mind renaming it already to `PACKED`.

the only reason not to rename things that are not completely off, is to keep changes to the blame log low, and avoid conflicts, but since we'll have to rename `ZIPLIST`, and since `NONE` ins unused, then i suppose we're going to change all the lines that refer to `container`, right?

if that's right, and we're going to change all lines anyway, then i don't mind to change `container` too, but if we rename it to `packed` then we should indeed use `UNPACKED` rather than `PLAIN`.

so both options are viable to me:
1. `unsigned int container : 2; / PLAIN==1 or PACKED==2 */`
2. `unsigned int packed : 2; / UNPACKED==1 or PACKED==2 */`

my previous aim was to take this:
3. `unsigned int container : 2; / PLAIN==1 or ZIPLIST==2 */`
i.e. in this case we don't modify the variable name or the constant that's currently used.
and then when @sundb makes the PR that changes ziplist to listpack, he can change ZIPLIST into either LISTPACK or PACKED.

i think i actually prefer 3, since it'll reduce the LOC change in this PR. it doesn't touch the ziplists, so why rename the define or variable?

bottom line, i vote for taking option 3 in this PR, and then change it option 2 in a future PR.

Reply:
Should we add a macro `NODE_IS_PLAIN`, instead of the existing `node->container == QUICKLIST_NODE_CONTAINER_NONE`.